src/test/clojure/xtdb/check_pbuf.clj

```diff
@@ -0,0 +1,39 @@
+(ns xtdb.check-pbuf
+  (:require [clojure.string :as str]
+            [xtdb.test-json :as tj])
+  (:import [java.nio.file FileVisitOption Files Path]
+           (xtdb.block.proto Block TableBlock)))
+
+(defn multi-parse-fn
+  "Tries to parse a byte array from a given list of parse-fns. Throws if non parses."
+  [parse-fns]
+  (fn [& args]
+    (or (some #(try (apply % args) (catch Exception _e nil)) parse-fns)
+        (throw (RuntimeException. "No matching parse-fn!")))))
+
+(defn check-pbuf
```

Comment:
we can probably make this a bit simpler than the JSON equivalent - we're looking at bringing a flat directory full of binpbs into a Clojure map, and comparing them against a known dir

Reply:
I am not following. We have two directories with `binpb`s in them. How should I compare them?